src/main/java/org/opentripplanner/routing/graph/index/StreetIndex.java

```diff
@@ -435,4 +463,18 @@ private static Map<FeedScopedId, TransitStopVertex> toImmutableMap(
     }
     return Map.copyOf(map);
   }
+
+  private static Map<FeedScopedId, TransitStationCentroidVertex> createTransitStationVertexMap(
+    Graph graph
+  ) {
+    var vertices = graph.getVerticesOfType(TransitStationCentroidVertex.class);
```

Comment:
We are doing the same for the stop vertices. If it's an issue I can probably build both maps at the same time so we don't have to iterate twice

Reply:
On my laptop this takes about 100ms for our dataset of 1.5M vertices. Since it's only done once during server startup, I don't think it's much to worry about?

Reply:
The way to solve this is not to iterate over the station once and throw everything that need to done with them in the same loop, but to cashe the set of stop/station vertexes during graph build. I am not sure if we wan to keep these indexes around during routing, but if not the we need to split the data structure used for building and routing - probably a good idea anyway. I am not worried about the performance here, so this is probably something we can do later.

Reply:
Germany has 23 million vertices so it would take about 1.5 seconds during start up. We have bigger fish to fry.